src/isoStringScalar/isoStringScalar.ts

```diff
@@ -0,0 +1,86 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import { isValid, parseISO, toDate } from 'date-fns';
+import {
+  InternalServerError,
+  UserInputError,
+} from '../errorHandler/errorHandler';
+
+export const isoStringScalar = new GraphQLScalarType({
+  name: 'ISOString',
+  description: 'ISOString custom scalar type',
+
+  /**
+   * Converts server-side TS Date representation to a JSON-compatible, ISO-8601-compliant
+   * UTC Datetime String for Apollo Server to include in an operation response.
+   *
+   * @param value - TS Date Object as generated by Data Store Client & validated by mysqlDateConvert.
+   * @returns ISO-8601-compliant, UTC-based Datetime String.
+   */
+  serialize(value: Date | null): string | null {
+    if (value === null) {
+      return null;
+    }
+
+    if (value instanceof Date && isNaN(value.valueOf())) {
+      throw new InternalServerError(
+        'Invalid Data Store Response: invalid Date object'
+      );
+    }
+
+    if (value instanceof Date) {
+      // toISOString() is always zero UTC offset per
+      // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
+      const resp: string = value.toISOString();
+      return resp;
+    }
+
+    throw new InternalServerError(
+      'GraphQL ISOString Scalar serializer expected a `Date` object or null'
+    );
+  },
+
+  /**
+   * Converts the scalar's JSON value to its back-end representation
+   * before it's added to a resolver's args.
+   *
+   * @param value - ISO-8601-compliant Datetime String.
+   * @returns TS Data Object for interacting with Data Store via Client.
+   */
+  parseValue(value: string | null): Date | null {
+    if (value === null || value === '') {
+      return null;
+    }
+
+    if (typeof value === 'string') {
+      const dt = parseISO(value);
+
+      if (!isValid(dt)) {
+        throw new UserInputError(
+          'Invalid User Input: ISOString Scalar parse expected a ISO-8601-compliant string'
+        );
+      }
+
+      return toDate(dt);
+    }
+
+    throw new UserInputError(
+      'Invalid User Input: ISOString Scalar parse expected a value of type string or null'
+    );
+  },
+
+  /**
+   * Converts the value's AST representation to the scalar's back-end representation.
+   *
+   * @param ast - AST Representation of ISOString.
+   * @returns TS Date Object.
+   */
+  parseLiteral(ast): Date | null {
+    if (ast.kind === Kind.STRING) {
```

Comment:
Ditto optional style: invert this and return an error if it is not `Kind.STRING`.

I think we also want a check that the parsed date is valid and not NaN here.

Reply:
makes sense. I also think bc I'm just making sure the AST kind is string then grabbing the AST value & passing it to parseValue, parseValue will confirm the NaN check via date-fns isValid (0000-00-00 etc. isn't valid ISO-8601).

I also inverted the parseValue function to for the desired type to be on the happy path.

Reply:
You're right, I definitely misread this before.